### PR TITLE
Fix #201 correcly calculate the position of struct base expression span.

### LINKED ISSUE #201: Unexpected coment added with struct syntax
Given the program:

```rust
fn main() {
    let state = get_state();
    let loop_body_state = ExecutionState { instr_ptr: 0, .. state.clone() };
}
```

Rustfmt generates the program:

```rust

fn main() {
    let state = get_state();
    let loop_body_state = ExecutionState { instr_ptr: 0 /* . */, ..state.clone() };
}
```

I don't think that comment should be there.

### PR DESCRIPTION
**Pull request by sinhpham**

### CODE REVIEW

File: src/expr.rs [hunk 1]
```diff
@@ -964,7 +964,14 @@ fn rewrite_struct_lit<'a>(context: &RewriteContext,
...
                                      StructLitField::Regular(ref field) => field.span.lo,
                                      // 2 = ..
-                                     StructLitField::Base(ref expr) => expr.span.lo - BytePos(2),
+                                     StructLitField::Base(ref expr) => {
+                                         let snippet =
+                                             context.codemap.span_to_snippet(mk_sp(span_after(span,
+                                             "{", context.codemap), expr.span.lo)).unwrap();
+                                         let pos = snippet.find_uncommented("..").unwrap();
+                                         span_after(span, "{", context.codemap) +
+                                         BytePos(pos as u32)
```
Comment (cassiersg) [inline src/expr.rs]: The indentation is off here.

**Comment by cassiersg** [comment]
Thanks for the PR !

I think the fix will not work if there is another `..` in the struct :
```rust
A{ a: B{..b}, .. c}
```
The search span should start at the end of the last field.

**Comment by sinhpham** [comment]
Hmm you're right, seems like span_after(span, "{", context.codemap) doesn't do what I thought. How can I access the end position of the last field? Add a last_item parameter to F1 get_lo (this seems invasive)? Or can we take another approach and solve this problem in ListItems iterator (having peek() to look ahead but currently don't know about the StructLitField type).

**Comment by cassiersg** [comment]
I think replacing `span` at line 696 with something like (untested)
```rust
mk_sp(fields.last().map_or(span.lo, |field| field.span.hi), expr.span.lo)
```
should do the job.

**Comment by sinhpham** [comment]
Awesome, it seems to work, PR recreated.

**Pull request closed without merging**